homeassistant/components/sensor/recorder.py

```diff
@@ -134,16 +134,7 @@ def _time_weighted_average(
         duration = end - old_start_time
         accumulated += old_fstate * duration.total_seconds()
 
-    period_seconds = (end - start).total_seconds()
-    if period_seconds == 0:
-        # If the only state changed that happened was at the exact moment
-        # at the end of the period, we can't calculate a meaningful average
-        # so we return 0.0 since it represents a time duration smaller than
-        # we can measure. This probably means the precision of statistics
-        # column schema in the database is incorrect but it is actually possible
-        # to happen if the state change event fired at the exact microsecond
-        return 0.0
-    return accumulated / period_seconds
```

Comment:
This was introduced in PR https://github.com/home-assistant/core/pull/90069 to fix https://github.com/home-assistant/core/issues/88846

Or do we want to keep it around to guard against databases with single precision float timestamp columns in case they've done some botched import of SQLite?
I think that's not needed though because single precision columns is one of the error we check for here: <https://github.com/home-assistant/core/blob/0fd2a7983d6c2dc84cb829575e9871d12cdf3a9a/homeassistant/components/recorder/migration.py#L299-L307>

Reply:
I think this could still happen if they botch a migrate and the data in the database has lost the time precision because once its been truncated fixing the column won't bring back lost precision.

Reply:
per discussion on discord, we think this was cause by the lack of timestamp check in #56785 which could lead to the same state being seen here